### DISCUSSION TITLE
branches

### DISCUSSION THREAD

**Discussion by bigclawBin** [Rapptz/discord.py#9696] [Q&A] [answered]
i don`t now how create branches
`            channel = bot.get_channel(int(ctx.author.id))
            thread = await channel.create_thread(
                name="example",
                type=ChannelType.public_thread
            )
            await thread.send("This is an example message")`
 but Error  
```
thread = await channel.create_thread(
                   ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'create_thread'
```

**Comment by tailoric** [comment, marked as answer]
`get_channel` returns None if the bot can't find the channel with that id and you are trying to find a channel using a user id.
If you want the channel the command was used in then just use `ctx.channel`

**Reply by bigclawBin** [reply]
thanks

**Reply by bigclawBin** [reply]
And how to make a branch related to the user?

**Reply by tailoric** [reply]
This is difficult to answer because I have no idea what you mean with making a thread (I assume that is what you mean with branch) related to a user.

The only ways a user can be related to a thread is by being the owner of a thread or by being added to the thread.
If your bot is creating the thread then it will be the owner of the thread not the user invoking the command. 

If you mean adding the user then you can do it via the thread instance and calling [`thread.add_user` ](https://discordpy.readthedocs.io/en/latest/api.html?highlight=add_user#discord.Thread.add_user)
so something like this:

```py
thread = await ctx.channel.create_thread(name="example", type=discord.ChannelType.public_thread)
await thread.add_user(ctx.author)
```

You need to be more specific however since I have no idea what you mean with your question.
